app/game/game.js

```diff
@@ -72,22 +77,50 @@ const GAME = (humanUsername, aiUsername, gameId) => {
 
     const spawnThreat = () => {
         const threatRoom = selectThreatRoom(AVAILABLE_ROOMS, ROOMS_WITH_THREATS);
-        const threat = Threat(() => onThreatUnresolved(threatRoom, THREATS.length));
-        THREATS.push(threat);
+        const threat = Threat(randomlySelectThreat(), () => onThreatUnresolved(threatRoom, THREATS.length));
```

Comment:
When the threat is unresolved, it removes the threat at the index `THREATS.length`. The assumption is that the index of this threat won't change between now and when it is removed. But this is not true. An earlier threat in the array may be removed, meaning the index has changed and the index now corresponds to the wrong threat.

Reply:
Now threat is indexed by room